src/mscorlib/src/System/String.cs

```diff
@@ -973,17 +973,20 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
 
             if ((count == 0) || (omitEmptyEntries && this.Length == 0)) 
             {           
-                return new String[0];
+                return EmptyArray<String>.Value;
+            }
+
+            if (count == 1)
+            {
+                return new String[] { this };
             }
             
             int[] sepList = new int[Length];            
             int numReplaces = MakeSeparatorList(separator, ref sepList);            
             
             //Handle the special case of no replaces and special count.
-            if (0 == numReplaces || count == 1) {
-                String[] stringArray = new String[1];
-                stringArray[0] = this;
-                return stringArray;
+            if (0 == numReplaces) {
```

Comment:
The comment needs to be updated (or separated to move the count part up to where the count is now checked).